js/client.js

```diff
@@ -806,12 +813,17 @@
 				var roomTypeLFIndex = roomType.indexOf('\n');
 				if (roomTypeLFIndex >= 0) roomType = roomType.substr(0, roomTypeLFIndex);
 				roomType = toId(roomType);
+
 				if (this.rooms[roomid] || roomid === 'staff') {
 					// staff is always joined in background
 					this.addRoom(roomid, roomType, true);
 				} else {
 					this.joinRoom(roomid, roomType, true);
 				}
+
+				if (roomType === 'chat' && roomid !== 'staff') {
+					this.addAutojoin(roomid, true);
```

Comment:
The difference in performance is really not worth it: remember that `sliceAutojoins` will loop a maximum of six times and is basically O(1). More important is the difference in complexity and brittleness (same reason why I said to use `>= 6` instead of `=== 6`).